plugin/plugin.go

```diff
@@ -569,22 +575,32 @@ func awsInstanceToHost(instance *ec2.Instance) (*pb.ListHostsResponseHost, error
 				return nil, errors.New("response integrity error: interface address entry is nil")
 			}
 
-			// Check to see if the PrivateIpAddress matches the one
-			// reported at the top-level of the instance (let's call it the
-			// "default address"). If it doesn't, add it.
+			// Check to see if the PrivateIpAddress/PrivateDnsName fields
+			// match the ones reported at the top-level of the instance.
+			// If they don't, add them.
 			if aws.StringValue(addr.PrivateIpAddress) != "" {
 				if aws.StringValue(instance.PrivateIpAddress) != aws.StringValue(addr.PrivateIpAddress) {
 					result.IpAddresses = append(result.IpAddresses, aws.StringValue(addr.PrivateIpAddress))
 				}
 			}
+			if aws.StringValue(addr.PrivateDnsName) != "" {
```

Comment:
This is the private dns name for the private ip address on this iface.   The iface itself also has a private dns name... should that be included as well?

Reply:
I'm not too sure. I think that this would just be the default address/dns name for the interface itself, and would be duplicated in the addresses... but given that, maybe we should do the same thing that we do for the instance. I'll work on it.

Reply:
Just confirmed that the addresses do seem to follow this behavior (although only just did it with a cursory check with a single address). So the same kind of logic just nested up a level should work.

Reply:
Now done. I've also moved the duplicate check to just a rudimentary check of the whole current set of IP addresses/dns names.